astro/src/content/blog/llms-for-docs.mdx

```diff
@@ -0,0 +1,231 @@
+
+### Success Metrics
+
+Are these files downloaded? How often? By what user agents? Is it being primarily downloaded by LLMs or users?
+
+The answers to these questions lie in your access logs, since JavaScript tools like Google Analytics don't track direct requests to files.
+
+## Enable Developer Self-Service with a Copy-to-Markdown Button
+
+Adding a "copy to markdown" button like below helps developers leverage LLMs in a similar way to providing the `llms-full.txt` file mentioned above. Folks can copy the text and paste it into their favorite LLM, while using fewer tokens than the `llms-full.txt` file.
+
+We've installed this on every page on our docs site; an example is below:
+
+![A copy to markdown button on a FusionAuth documentation page.](/img/blogs/llms-for-docs/markdown-button.png)
+
+By offering this button, you let developers transform your documentation for their purpose. Here are some examples of prompts a developer might do for a technical doc like an API:
+
+* summarize all the error statuses of this API
+* explain the key concepts here to me "like I'm 5" or "like I'm an expert software architect" or "like I'm a front end developer"
+* how does this API compare with a similar API from \[competitor]
+* show me how to use this API in Rust or C++ (or other possibly unsupported languages)
+* provide me 10 sets of fake API requests for this API to use in my test harness
+* translate this page to Spanish or Swedish
+
+Basically, you don't know exactly what is important to the developer at that moment in time, but they do. And they can ask the LLM to provide it.
+
+Of course someone could copy the HTML and then do the transformation, but that uses more tokens and is more cumbersome.
+
+### Implementation Notes
+
+We built JavaScript to perform the copy operation:
+
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-site/refs/heads/main/astro/src/tools/docs/copyMarkdownToClipboard.js" lang="javascript" />
+
+We also used an astro `post:build` plugin to generate the markdown file at a sibling location to the HTML, so the JavaScript above could retrieve it.
+
+<RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-site/refs/heads/main/astro/src/plugins/markdown-extract.js" lang="javascript" />
+
+This astro plugin code:
+
+* finds all the document pages in a collection
+* inlines all the content in `src/diagrams` and `src/content` (we aren't as worried about overflowing the context window with this single page of markdown)
+* removes all the frontmatter
+* writes a peer markdown file in a specific location
+
+There are still component tags in the output markdown because components provide functionality and HTML. There are alternative solutions that take HTML and turn it into pure markdown we may use in the future.
```

Comment:
`There are alternative solutions that take HTML and turn it into pure markdown we may use in the future.` -> "We may use alternative solutions that take HTML and turn it into pure markdown in the future."